superset/databases/dao.py

```diff
@@ -0,0 +1,75 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import logging
+from contextlib import closing
+
+from flask import g
+from sqlalchemy import select
+
+from superset import app
+from superset.dao.base import BaseDAO
+from superset.extensions import db
+from superset.models.core import Database
+from superset.security.analytics_db_safety import check_sqlalchemy_uri
+
+logger = logging.getLogger(__name__)
+
+
+class DatabaseDAO(BaseDAO):
```

Comment:
This probably makes more sense as a Command rather than a DAO since it is more of a business operation.